### PR TITLE
Add concept BDD tests and remove old integration tests

### PR DESCRIPTION
**Pull request by adammitchelldev**
## What is the goal of this PR?

We are now going to use verification's BDD tests to test client-java concept API, rather than rolling our own concept integration tests. This gives us more complete test coverage.

## What are the changes implemented in this PR?

- Merged some things from master.
- Added BDDs (which are currently failing) and refactor to client-java so that they compile.
- Updated to use new 2.0 test runner from common.
- Updated to use private snapshot repo for grakn 2.0 artifact.

### CODE REVIEW

File: test/behaviour/connection/session/SessionSteps.java [hunk 1]
```diff
@@ -40,27 +43,61 @@
...
+    @When("connection open (data )session(s) for database(s):")
+    public void connection_open_data_sessions_for_databases(List<String> names) {
         for (String name : names) {
             sessions.add(client.session(name));
         }
     }
 
-    @When("connection open sessions in parallel for keyspaces:")
-    public void connection_open_sessions_in_parallel_for_keyspaces(List<String> names) {
+    @When("connection open (data )sessions in parallel for databases:")
```
Comment (flyingsilverfin) [inline test/behaviour/connection/session/SessionSteps.java:70]: will this work if there's a mismatch in the spacing in the BDD scenario definition?

File: test/integration/tracing/TracingIT.java [hunk 2]
```diff
@@ -1,81 +0,0 @@
-/*
```
Comment (flyingsilverfin) [inline test/integration/tracing/TracingIT.java:1]: so how come this test is gone as well? Now we have not tracing tests...

File: test/integration/answer/AnswerIT.java [hunk 3]
```diff
@@ -1,332 +0,0 @@
-/*
```
Comment (flyingsilverfin) [inline test/integration/answer/AnswerIT.java:1]: why is this test gone now? We don't have any tests for Answer now right? If we don't want it get we should @ignore it and then add it to the issue you're making with all the tests to re-enable

File: GraknClient.java [hunk 4]
```diff
@@ -634,6 +634,15 @@ public void commit() {
             close();
         }
 
+        public ThingType.Remote<?, ?> getRootType() {
+            grakn.client.concept.type.Type.Remote<?> concept = getType(Label.of("thing")); // TODO do this properly
```
[the remark this answers is not in the document]
Comment (adammitchelldev) [reply GraknClient.java:638]: Not yet, MetaTypes work very differently on hypergraph and client-java and this is not in the scope of this PR, the idea of thing as the root type for things is actually different to how it was before.

Also, the existing client-java code did just this, I added the TODO only I think.

File: test/behaviour/connection/session/SessionSteps.java (hunk 1 above)
Comment (adammitchelldev) [reply test/behaviour/connection/session/SessionSteps.java:70]: no

File: test/integration/answer/AnswerIT.java (hunk 3 above)
Comment (adammitchelldev) [reply test/integration/answer/AnswerIT.java:1]: All of the integration tests were out of date and would be unbuildable, not just untestable. They can be migrated in at a later time right? They still exist in master.

**Comment by adammitchelldev** [comment]
I have created issues for the 2.0 TODOs, which Alex will probably move on to implementing.

File: GraknClient.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (adammitchelldev) [reply GraknClient.java:638]: Okay, more specifically, this method is new, but this is the only way client-java is able to get types and is the same method as the other meta/root type getters.

File: test/integration/answer/AnswerIT.java (hunk 3 above)
Comment (flyingsilverfin) [reply test/integration/answer/AnswerIT.java:1]: ok good thanks for making issue for these

File: test/behaviour/connection/session/SessionSteps.java (hunk 1 above)
Comment (flyingsilverfin) [reply test/behaviour/connection/session/SessionSteps.java:70]: So where is the typo - here `(data )` ? or In verification do we need to change it to be correct `(data)`?

**Pull request merged**